### PR TITLE
Issue a warning on --with-dns option to proxy, because it was removed

### LINKED ISSUE #1810: Warn, don't fail, on `--with-dns` option to the proxy
#1754 was a bit heavy-handed; the option should remain with a warning message

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #1810

### CODE REVIEW

File: prog/weaveproxy/main.go [hunk 1]
```diff
@@ -39,6 +40,7 @@ func main() {
 	mflag.BoolVar(&c.TLSConfig.Enabled, []string{"#tls", "-tls"}, false, "Use TLS; implied by --tlsverify")
 	mflag.StringVar(&c.TLSConfig.Key, []string{"#tlskey", "-tlskey"}, "", "Path to TLS key file")
 	mflag.BoolVar(&c.TLSConfig.Verify, []string{"#tlsverify", "-tlsverify"}, false, "Use TLS and verify the remote")
+	mflag.BoolVar(&withDNS, []string{"#-with-dns", "#w"}, false, "option removed")
```
Comment (rade) [inline prog/weaveproxy/main.go]: `"(deprecated)"`

File: prog/weaveproxy/main.go [hunk 2]
```diff
@@ -57,6 +59,10 @@ func main() {
 	Log.Infoln("weave proxy", version)
 	Log.Infoln("Command line arguments:", strings.Join(os.Args[1:], " "))
 
+	if withDNS {
+		Log.Warning("--with-dns option has been removed; DNS is on by default")
```
Comment (rade) [inline prog/weaveproxy/main.go]: "...has been deprecated; ..."

Comment (rade) [inline prog/weaveproxy/main.go]: in fact I'd make the message identical to that in the script, i.e. "--with-dns is deprecated; it is on by default".

Comment (bboreham) [inline prog/weaveproxy/main.go]: "deprecated" suggests the behaviour is still there, but will be removed later.  This is not what happened.

Comment (rade) [inline prog/weaveproxy/main.go]: Then why does the script say "deprecated"?

Comment (bboreham) [inline prog/weaveproxy/main.go]: AFAIK that's a different `--with-dns` option, on `weave run`, where the behaviour did not change.

Comment (rade) [inline prog/weaveproxy/main.go]: No, I am not talking about `weave run` but the `--with-dns` case you added to `deprecation_warnings` in this very PR.

Comment (bboreham) [inline prog/weaveproxy/main.go]: Ah, overdid the copy'n'paste

Comment (bboreham) [inline prog/weaveproxy/main.go]: I fixed that message to say "removed".

**Pull request merged**